tooltester/src/main/java/io/dockstore/tooltester/client/cli/Client.java

```diff
@@ -130,48 +204,318 @@ protected void setupClientEnvironment() {
+                        }
+                }
+            } catch (ApiException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Creates the pipeline(s) on Jenkins to test a tool
+     *
+     * @param tool  The tool to get the test results for
+     */
+    void getToolTestResults(Tool tool) {
+        DockstoreTool dockstoreTool = null;
+        try {
+            dockstoreTool = getContainersApi().getPublishedContainerByToolPath(tool.getId());
+        } catch (ApiException e) {
+            e.printStackTrace();
+        }
+        List<ToolVersion> toolVersions = tool.getVersions();
+        for (ToolVersion toolversion : toolVersions) {
+            Long containerId = dockstoreTool.getId();
+            String id = toolversion.getId();
+            String tag = toolversion.getName();
+            String name = id;
+            name = name.replace("/","-");
+            name = name.replace(":","-");
+            Date date = new Date();
+            String dockerfileStatus = dockerfileTester.getTestResults(name);
+            if (!dockerfileStatus.equals("SUCCESS")) {
+                dockerfileStatus = dockerfileStatus + " See " + dockerfileTester.createConsoleOutputFile(name);
+            }
+                try {
+                    List<SourceFile> testParameterFiles;
+                    for (ToolVersion.DescriptorTypeEnum descriptorType : toolversion.getDescriptorType()) {
+                            testParameterFiles = containersApi.getTestParameterFiles(containerId, tag, descriptorType.toString());
+                            for (SourceFile testParameterFile : testParameterFiles) {
+                                count++;
+                                String path = testParameterFile.getPath();
+                                path = path.replaceFirst("^/", "");
+                                path = path.replace("/", "-");
+                                String status = parameterFileTester.getTestResults(name + path);
+                                if (!status.equals("SUCCESS")) {
+                                    status = status + " See " + parameterFileTester.createConsoleOutputFile(name + path);
+                                }
+                                List<String> record = Arrays
+                                        .asList(String.valueOf(count), date.toString(), toolversion.getId(), tag, "Local", testParameterFile.getPath(), null, status, dockerfileStatus);
+                                report.writeLine(record);
+                            }
+                    }
+                } catch (ApiException e) {
+                    e.printStackTrace();
+                }
+
+        }
+    }
+
+    /**
+     * This function deletes all jobs on jenkins matching "Test.*"
+     */
+    void deleteJobs(String pattern) {
+        try {
+            Map<String, Job> jobs = jenkins.getJobs();
```

Comment:
This can be better represented as a lambda expression 
https://www.mkyong.com/java8/java-8-filter-a-map-examples/